Number_types/include/CGAL/boost_mp.h

```diff
@@ -223,7 +242,163 @@ struct RET_boost_mp <NT, boost::mpl::int_<boost::multiprecision::number_kind_int
 
 template <class NT>
 struct RET_boost_mp <NT, boost::mpl::int_<boost::multiprecision::number_kind_rational> >
-    : RET_boost_mp_base <NT> {};
+    : RET_boost_mp_base <NT> {
+    typedef NT Type;
+    struct To_interval
+        : public CGAL::cpp98::unary_function< Type, std::pair< double, double > > {
+
+        bool are_bounds_correct( const double l, const double u, const Type& x ) const {
+
+          const double inf = std::numeric_limits<double>::infinity();
+          CGAL_assertion(u == l || u == std::nextafter(l, +inf));
+          const bool are_bounds_tight = (u == l || u == std::nextafter(l, +inf));
+
+          if (
+            CGAL::abs(l) == inf ||
+            CGAL::abs(u) == inf ||
+            CGAL::abs(l) == 0.0 ||
+            CGAL::abs(u) == 0.0) {
+            return are_bounds_tight;
+          }
+          CGAL_assertion(CGAL::abs(l) != inf);
+          CGAL_assertion(CGAL::abs(u) != inf);
+          CGAL_assertion(CGAL::abs(l) != 0.0);
+          CGAL_assertion(CGAL::abs(u) != 0.0);
+
+          const Type lb(l), ub(u);
+          CGAL_assertion(lb <= x);
+          CGAL_assertion(ub >= x);
+          const bool are_bounds_respected = (lb <= x && x <= ub);
+          return are_bounds_tight && are_bounds_respected;
+        }
+
+        Interval_nt<false>
+        my_ldexp( const Interval_nt<false>& intv, const int e ) const {
```

Comment:
Why are we duplicating this function? (several times!)

Reply:
True. I refactored the duplicated code.

Reply:
We already have a function `ldexp` in Interval_nt.h. If you don't use it, you need to add a comment before `my_ldexp` explaining why this alternate function is useful, how it differs from the original function. Does it use positivity to skip some computations and become faster? Does it avoid setting the rounding mode? Does it fix some bug in the original function (why not fix it there then?)?
my_ldexp contains `double` multiplications without protection. Are we sure these multiplications always do the right thing whatever the rounding mode? Or that the rounding mode will always be the right one (an assertion would help document that)?